velox/connectors/hive/HiveConfig.h

```diff
@@ -31,8 +31,14 @@ class HiveConfig {
   static constexpr const char* kInsertExistingPartitionsBehavior =
       "insert_existing_partitions_behavior";
 
+  /// Maximum number of partitions per writer.
```

Comment:
Hi Masha, yes, by "per writer" it means. "per a single instance of TableWriter operator" (or thread). I will change the comment and configs.rst. I copied the text from Java, and this is how I understand it for now. 

We're replicating this config from Java. Regarding whether it's 100 partitions per Task or per thread, my understanding for now is per thread, so the theoretic total of 500 partitions is true assuming my understanding is correct. The following is the code flow, in case you have time to help double check.

This config is validated in class HivePageSink. (`pagePartitioner` is like our `PartitionIdGenerator` class that is applied across inputs)
https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java#L470-L473

`pagePartitioner` is per `HivePageSink` instance which is per `TableWriterOperator` instance. I don't find `pageSink` field of `TableWriterOperator` class is shared among drivers/threads.
https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java#L137-L147,
https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java#L341

`LocalExecutionPlanner` class is where the number of drivers for `TableWriterOperator` is set.
https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java#L2590-L2595

Reply:
@gggrace14 Ge, thank you for checking. I'm seeing that number of TableWriter operator instances is controlled by a session property task_partitioned_writer_count which defaults to the value of another session property task_writer_count.

```
                new PropertyMetadata<>(
                        TASK_PARTITIONED_WRITER_COUNT,
                        "Number of writers per task for partitioned writes. If not set, the number set by task.writer-count will be used",
                        BIGINT,
                        Integer.class,
                        taskManagerConfig.getPartitionedWriterCount(),
                        false,
                        value -> validateValueIsPowerOfTwo(value, TASK_PARTITIONED_WRITER_COUNT),
                        value -> value),
```

, but there is also a comment that suggests that writes are serialized. We'll need to dig in to understand the behavior better.

```
        @Override
        public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPlanContext context)
        {
            // Set table writer count
            if (node.getTablePartitioningScheme().isPresent()) {
                context.setDriverInstanceCount(getTaskPartitionedWriterCount(session));
            }
            else {
                context.setDriverInstanceCount(getTaskWriterCount(session));
            }

            // serialize writes by forcing data through a single writer
            PhysicalOperation source = node.getSource().accept(this, context);
```